packages/theme-doc/src/Layout/Footer.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { useContext, useMemo } from 'react'
+import RcFooter from 'rc-footer'
+
+import { themeConfigCtx, useThemeCtx } from '../ctx'
+import s from './Footer.module.less'
+
+export type FooterLink = {
```

Comment:
Could we allow users to pass options `icon`, `description`, `openExternal` of [rc-footer](https://www.npmjs.com/package/rc-footer)? I think they are useful to users.

![image](https://user-images.githubusercontent.com/18747423/190854905-cc225a8d-c0b2-442e-b20e-caf0a9f9c207.png)

Example usage: antd document:

https://github.com/ant-design/ant-design/blob/01087b2571ee26629b0df78e5a76c788572cebdb/site/theme/template/Layout/Footer.tsx#L46

Outcome:
![image](https://user-images.githubusercontent.com/18747423/190855056-a0a5f7b8-60b4-4d49-9820-d5520a6563c5.png)